tests/Mono.Android-Tests/Xamarin.Android.Net/AndroidMessageHandlerNegotiateAuthenticationTests.cs

```diff
@@ -0,0 +1,133 @@
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Xamarin.Android.Net;
+using NUnit.Framework;
+
+namespace Xamarin.Android.NetTests {
+	[TestFixture]
+	public sealed class AndroidMessageHandlerNegotiateAuthenticationTests
+	{
+		[SetUp]
+		public void SetUp ()
+		{
+			AppContext.SetSwitch ("Xamarin.Android.Net.UseNegotiateAuthentication", true);
+		}
+
+		[Test]
+		public async Task RequestWithoutCredentialsFails ()
+		{
+			if (!NegotiateAuthenticationIsEnabled) {
+				Assert.Ignore ("Negotiate authentication is not enabled.");
+			}
+
+			using var server = new FakeNtlmServer (port: 47662);
+			var handler = new AndroidMessageHandler ();
+			var client = new HttpClient (handler);
+
+			var response = await client.GetAsync (server.Uri);
+
+			Assert.IsFalse (response.IsSuccessStatusCode);
+			Assert.AreEqual (HttpStatusCode.Unauthorized, response.StatusCode);
+		}
+
+		[Test]
+		public async Task RequestWithCredentialsSucceeds ()
```

Comment:
Is there a "complete" NTLM test suite somewhere?  Something that checks for "invalid" hostnames, "valid but dodgy-looking" hostnames, escape characters, `../../../` path escape fragments, etc.?  Something to help verify that there are no latent security-related or -adjacent issues in this NTLM authentication?


"Obviously" running tests against a "fake NTLM server" is *not* a complete "real-world" unit test…